### PR TITLE
Add event endpoints - Closes #7126

### LINKED ISSUE #7126: Add `chain_getEvents` and `chain_proveEvents` endpoints
### Description
Add framework endpoints as below

##### Req
```
{
  height: number,
}
```
##### Res
```
{
  events: {
    moduleID: number,
    typeID: number,
    topics: string[],
    data: string,
    index: number,
  }[],
}
```
If height does not exist, it should return
```
{
  error: {
    message: `Events at height ${height} is not stored in the node.`,
  }
}
```
##### Req
```
{
  height: number,
  queries: string[],
}
```
##### Res
```
{
  siblingHashes: string[],
  queries: {
    key: string,
    value: string,
    bitmap: string,
  }[]
}
```
If height does not exist, it should return
```
{
  error: {
    message: `Events at height ${height} is not stored in the node.`,
  }
}
```

### Acceptance Criteria
- All endpoints are unit tested

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #7126

### How was it solved?

- Add new endpoints in node endpoint

### How was it tested?

- Add unit tests

### Additional Information
- Namespace will be updated in #6947

### CODE REVIEW

**Review by mehmetegemen — APPROVED** [review]

File: elements/lisk-tree/src/index.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import { MerkleTree } from './merkle_tree/merkle_tree';
 import { calculateMerkleRoot, calculateMerkleRootWithLeaves } from './merkle_tree/utils';
 import { SparseMerkleTree } from './sparse_merkle_tree/sparse_merkle_tree';
 import { verify, calculateRoot } from './sparse_merkle_tree/utils';
+import { Proof as SMTProof, Query as SMTQuery } from './sparse_merkle_tree/types';
```
Comment (ishantiw) [inline elements/lisk-tree/src/index.ts:21]: maybe in future we can already export SMTProof from lisk-tree as we rename it anyways

File: framework/src/node/endpoint.ts [hunk 2]
```diff
@@ -32,6 +33,25 @@ interface InitArgs {
 	registeredModules: BaseModule[];
 }
 
+const proveEventsRequestSchema = {
+	$id: '/node/endpoint/proveEventsRequestSchema',
+	type: 'object',
+	required: ['height', 'queries'],
```
Comment (ishantiw) [inline framework/src/node/endpoint.ts:39]: fieldNumber is missing

Comment (ishantiw) [inline framework/src/node/endpoint.ts:39]: fieldNumber is missing

File: framework/test/unit/node/endpoint.spec.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import { Event, StateStore } from '@liskhq/lisk-chain';
+import { getRandomBytes } from '@liskhq/lisk-cryptography';
+import { InMemoryKVStore } from '@liskhq/lisk-db';
+import { ModuleEndpointContext } from '../../../src';
+import { Endpoint } from '../../../src/node/endpoint';
+import { createImmutableAPIContext } from '../../../src/node/state_machine';
+import { nodeConfig } from '../../utils/configs';
+import { fakeLogger } from '../../utils/node';
+
+const createContext = (
```
Comment (ishantiw) [inline framework/test/unit/node/endpoint.spec.ts:24]: can we use some existing createContext funcs in testing or add this one to make it reusable

Comment (ishantiw) [inline framework/test/unit/node/endpoint.spec.ts:24]: can we use some existing createContext funcs in testing or add this one to make it reusable

**Review by ishantiw — COMMENTED** [review]
LGTM, just a few comments

File: framework/src/node/endpoint.ts (hunk 2 above)
Comment (shuse2) [reply framework/src/node/endpoint.ts:39]: Endpoint req/res are not encoded(ie: standard JSON schema). Therefore, fieldNumber is not required and it shouldnt use dataType

File: elements/lisk-tree/src/index.ts (hunk 1 above)
Comment (shuse2) [reply elements/lisk-tree/src/index.ts:21]: yes, but the SMT types will be provided by lisk-db, so I will update there

**Review by ishantiw — APPROVED** [review]
🚀

**Pull request merged**